src/ServiceControl/CustomChecks/CustomChecksComponent.cs

```diff
@@ -24,29 +21,15 @@ public override void Configure(Settings settings, IHostBuilder hostBuilder)
                 serviceCollection.AddEventLogMapping<CustomCheckSucceededDefinition>();
                 serviceCollection.AddPlatformConnectionProvider<CustomChecksPlatformConnectionDetailsProvider>();
                 serviceCollection.AddSingleton<CustomCheckResultProcessor>();
-
-                switch (settings.DataStoreType)
-                {
-                    case DataStoreType.InMemory:
-                        serviceCollection.AddSingleton<ICustomChecksDataStore, InMemoryCustomCheckDataStore>();
-                        break;
-                    case DataStoreType.RavenDb:
-                        serviceCollection.AddSingleton<ICustomChecksDataStore, RavenDbCustomCheckDataStore>();
-                        break;
-                    case DataStoreType.SqlDb:
-                        serviceCollection.AddSingleton<ICustomChecksDataStore>(sp => new SqlDbCustomCheckDataStore(settings.SqlStorageConnectionString, sp.GetService<IDomainEvents>()));
-                        break;
-                    default:
-                        break;
-                }
             });
         }
 
         public override void Setup(Settings settings, IComponentInstallationContext context)
         {
+            //TODO can we do this dynamically somehow?
```

Comment:
Do we still need this?

Reply:
Yes until we agree how to do the setup - currently it's in a separate file as a static function - trying to see if we can make it part of the persistence dll and load/run it dynamically.